newIDE/app/src/ProjectsStorage/LocalProjectWriter.js

```diff
@@ -80,6 +134,38 @@ export default class LocalProjectWriter {
     }
   };
 
+  static saveProjectAs = (project: gdProject): Promise<void> => {
+    const filepath = project.getProjectFile();
+    const filenameWithExtension = path.basename(project.getProjectFile()) || "game.json";
+    const projectPath = path.dirname(project.getProjectFile());
+    const fileSystem = assignIn(new gd.AbstractFileSystemJS(), localFileSystem);
+    const browserWindow = electron.remote.getCurrentWindow();
+    const options = {
+    defaultPath: path.join(projectPath, filenameWithExtension),
+    filters: [{ name: 'GDevelop 5 project', extensions: ['json'] }],
+    };
+
+    if (!dialog) {
+    return Promise.reject('Unsupported');
+    }
+    const newFilepath = dialog.showSaveDialog(browserWindow, options);
+    if (!newFilepath || newFilepath === '') {
+    return Promise.reject('Filepath from dialog is empty');
+    }
+    const newProjectFolder = path.dirname(newFilepath);
+
+    gd.ProjectResourcesCopier.copyAllResourcesTo(
+    project,
+    fileSystem,
+    newProjectFolder,
+    true,
+    false,
+    true
+    );
+
+    return writeProjectFiles(project, newFilepath, newProjectFolder);
```

Comment:
To communicate that the save was done until the end, return a true boolean:
(*See also my other comments about handling this boolean.)

```
  return writeProjectFiles(project, newFilepath, newProjectFolder).then(() => {
    return true; // Save was done successfully and not cancelled.
  })
```